autokeras/auto/image.py

```diff
@@ -0,0 +1,130 @@
+import tensorflow as tf
+
+from autokeras import const
+from autokeras.auto import auto_pipeline
+from autokeras.auto import processor
+from autokeras.hypermodel import hyper_block, hyper_node
+from autokeras.hypermodel import hyper_head
+from autokeras import tuner
+
+
+class ImageTuner(tuner.SequentialRandomSearch):
+
+    def _run(self, hyperparameters, fit_kwargs):
+        # Build a model instance.
+        model = self.hypermodel.build(hyperparameters)
+
+        # Optionally disallow hyperparameters defined on the fly.
+        old_space = hyperparameters.space[:]
+        new_space = hyperparameters.space[:]
+        if not self.allow_new_parameters and set(old_space) != set(new_space):
+            diff = set(new_space) - set(old_space)
+            raise RuntimeError(
+                'The hypermodel has requested a parameter that was not part '
+                'of `hyperparameters`, '
+                'yet `allow_new_parameters` is set to False. '
+                'The unknown parameters are: {diff}'.format(diff=diff))
+
+        datagen = tf.keras.preprocessing.image.ImageDataGenerator(
```

Comment:
Prefer using tf.data for a pipeline. Reference: https://www.tensorflow.org/tutorials/load_data/images

You'll probably need some basic utility for data augmentation with tf.data, let's touch base on that later.

Reply:
I will submit another pull request for changing this one.